devito/operator/profiling.py

```diff
@@ -260,14 +261,19 @@ def __init__(self, name):
             compiler.add_ldflags('-Wl,-rpath,%s' % libdir)
 
     def instrument(self, iet):
-        sections = FindNodes(Section).visit(iet)
-
-        # Transform the Iteration/Expression tree introducing Advisor calls that
-        # resume and stop data collection
-        mapper = {i: List(body=[Call(self._api_resume), i, Call(self._api_pause)])
-                  for i in sections}
-        iet = Transformer(mapper).visit(iet)
-
+        # Look for the presence of a time loop within the IET of the Operator
+        found = False
+        for node in FindNodes(Iteration).visit(iet):
+            if node.dim.is_Time:
+                found = True
+                break
+
+        if found:
+            # The calls to Advisor's Collection Control API are only for Operators with
+            # a time loop
+            return List(header=c.Statement('%s()' % self._api_resume),
+                        body=iet,
+                        footer=c.Statement('%s()' % self._api_pause))
```

Comment:
for homogeneity:

* either blank line before the final return statement
* or `else: ...`